### ISSUE TITLE
Error 2276 for Valid Zsh Assoc Array Assignment

### ISSUE THREAD

**Issue by wdeshazer** [koalaman/shellcheck#3087]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2148, SC2276, SC1036, SC1088
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
typeset -xgA ම

ම=( [ZSYS]=Something )
ම+=( [ZUSER]=Something\ Else )
```

#### Here's what shellcheck currently says:
```
Line 1	SC2148: Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive.
Line 3	SC2276: This is interpreted as a command name containing '='. Bad assignment or comparison?
Line 3	SC1036: '(' is invalid here. Did you forget to escape it?
Line 3	SC1088: Parsing stopped here. Invalid use of parentheses?
```
#### Here's what I wanted or expected to see:

1) Assignment of an associative array or an array can be done post declaration.
2) The '+=' syntax with additional key-value's is valid syntax.
https://zsh.sourceforge.io/Guide/zshguide05.html#5.4.2

Here is the relevant excerpt from the docs. Chap 15.2

[15 Parameters](https://zsh.sourceforge.io/Doc/Release/Parameters.html#Array-Parameters)

![image](https://github.com/user-attachments/assets/644ecbdd-65d0-42f8-a3e1-5e7ded7cd5b4)

Let me know if you need anything else.

**Comment by brother** [comment]
zsh is not among the supported shell dialects (sh, bash, dash, ksh, busybox), closing. See #809.
